paimon-core/src/main/java/org/apache/paimon/operation/FileStoreExpireImpl.java

```diff
@@ -88,6 +89,12 @@ public FileStoreExpireImpl(
             SnapshotManager snapshotManager,
             ManifestFile.Factory manifestFileFactory,
             ManifestList.Factory manifestListFactory) {
+        Preconditions.checkArgument(
+                numRetainedMin >= 1,
+                "The minimum number of completed snapshots to retain should be >= 1.");
+        Preconditions.checkArgument(
+                numRetainedMax > numRetainedMin,
+                "The maximum number of snapshots to retain should be > the minimum number.");
```

Comment:
I think the limit should be `numRetainedMax >= numRetainedMin` because we can set `numRetainedMax == numRetainedMin` to control exactly how many snapshots are retained.

Reply:
Yes you are correct. I was confused by the following code in snapshot expiration.

```java
        // find the earliest snapshot to retain
        for (long id = Math.max(latestSnapshotId - numRetainedMax + 1, earliest);
                id <= latestSnapshotId - numRetainedMin;
                id++) {
            if (snapshotManager.snapshotExists(id)
                    && currentMillis - snapshotManager.snapshot(id).timeMillis()
                            <= millisRetained) {
                // within time threshold, can assume that all snapshots after it are also within
                // the threshold
                expireUntil(earliest, id);
                return;
            }
        }
```

This segment tries to locate the first snapshot within a range of snapshots, specifically between the numRetainedMax th and numRetainedMin th latest snapshots. This snapshot needs to be preserved because it doesn't fulfill the time threshold condition for expiration.

I've updated the comment to be more clear.